src/pushprocessor.ts

```diff
@@ -174,9 +206,21 @@ export class PushProcessor {
 
         // Merge the client-level defaults with the ones from the server
         const globalOverrides = newRules.global.override;
-        for (const override of DEFAULT_OVERRIDE_RULES) {
+        for (let override of DEFAULT_OVERRIDE_RULES) {
             const existingRule = globalOverrides.find((r) => r.rule_id === override.rule_id);
 
+            // Dynamically add the user ID as the value for the is_user_mention rule.
+            if (override.rule_id === RuleId.IsUserMention) {
+                // If the user ID wasn't provided, skip the rule.
+                if (!userId) {
+                    continue;
+                }
+
+                override = JSON.parse(JSON.stringify(override)); // deep clone
+                const conditions = override.conditions!;
+                conditions[0].value = userId;
```

Comment:
Why not:

```
override.conditions?.[0].value = userId;
```

To me it's a little clearer that we are modifying override.

Reply:
Likely because I didn't realize you could use `?.` or `!.` in front of `[0]`.

Reply:
Yeah it is a little surprising, but useful here.